test/inference/gibbs_conditional.jl

```diff
@@ -0,0 +1,122 @@
+using Random, Turing, Test
+using Clustering
+
+dir = splitdir(splitdir(pathof(Turing))[1])[1]
+include(dir*"/test/test_utils/AllUtils.jl")
+
+
+@turing_testset "gibbs conditionals" begin
+    let N = 1000,
+        (α_0, θ_0) = (2.0, inv(3.0)),
+        λ_true = rand(Gamma(α_0, θ_0)),
+        σ_true = sqrt(1 / λ_true),
+        m_true = rand(Normal(0, σ_true)),
+        x = rand(Normal(m_true, σ_true), N)
```

Comment:
I'm curious, why a `let` block and not just regular assignments? Is it due to the other example below which uses variables of the same name? In this case two separate test sets might be an alternative.